deployments/common/ap-policy-definition.yaml

```diff
@@ -0,0 +1,874 @@
+                  - euc-jp
+                  - utf-8
+                  - gbk
+                  - windows-1251
+                  - big5
+                  - iso-8859-1
+                  - shift_jis
+                  - euc-kr
+                  - iso-8859-4
+                  - iso-8859-7
+                  - iso-8859-15
+                  type: string
+                blocking-settings:
+                  properties:
+                    evasions:
+                      items:
+                        properties:
+                          description:
+                            enum:
+                            - '%u decoding'
+                            - Apache whitespace
+                            - Bad unescape
+                            - Bare byte decoding
+                            - Directory traversals
+                            - IIS backslashes
+                            - IIS Unicode codepoints
+                            - Multiple decoding
+                            type: string
+                          enabled:
+                            type: boolean
+                          maxDecodingPasses:
+                            type: integer
+                        type: object
+                      type: array
+                    http-protocols:
+                      items:
+                        properties:
+                          description:
+                            enum:
+                            - Unparsable request content
+                            - Several Content-Length headers
+                            - 'POST request with Content-Length: 0'
+                            - Null in request
+                            - No Host header in HTTP/1.1 request
+                            - Multiple host headers
+                            - Host header contains IP address
+                            - High ASCII characters in headers
+                            - Header name with no header value
+                            - CRLF characters before request start
+                            - Content length should be a positive number
+                            - Chunked request with Content-Length header
+                            - Check maximum number of parameters
+                            - Check maximum number of headers
+                            - Body in GET or HEAD requests
+                            - Bad multipart/form-data request parsing
+                            - Bad multipart parameters parsing
+                            - Bad HTTP version
+                            - Bad host header value
+                            type: string
+                          enabled:
+                            type: boolean
+                          maxHeaders:
+                            description: optional
+                            type: integer
+                          maxParams:
+                            type: integer
```

Comment:
maxHeaders, maxParams, and evasions/maxDecoding passes are all "optional" as they only apply to one of the subviolation.